static/docs/user-guide/contributing.md

```diff
@@ -103,23 +103,166 @@ default `pytest`'s scope:
+$ pip install -e .[ssh]
+# or
+$ pip install -e .[all]
+```
+
+You will need to update your `ENV` throughout subsequent steps, so we created a
+template for you:
+
+```dvc
+$ cp tests/remotes_env.sample tests/remotes_env
+```
+
+Then uncomment lines you need and fill in needed values, the details are
+explained in remote specific subsections. To activate these env vars use:
+
+```dvc
+$ source tests/remotes_env
+```
+
+If some member of your team had already went through all of this you may just
+ask his `remotes_env` file and Google Cloud credentials and you can skip any
+manipulations with `ENV` below.
+
+<details>
+
+### Click for S3 testing instructions
+
+Install
+[aws cli](https://docs.aws.amazon.com/en_us/cli/latest/userguide/cli-chap-install.html)
+tools.
+
+Set up an account, get credentials, which will have access to S3. When set `ENV`
+vars like this:
+
+```dvc
+$ export AWS_ACCESS_KEY_ID="...YOUR-ACCESS-KEY-ID..."
+$ export AWS_SECRET_ACCESS_KEY="...YOUR-SECRET-ACCESS-KEY..."
+$ export DVC_TEST_AWS_REPO_BUCKET="...TEST-S3-BUCKET..."
+```
+
+</details>
+
+<details>
+
+### Click for Google Cloud storage testing instructions
+
+Go through [quickstart](https://cloud.google.com/sdk/docs/quickstarts) for your
+OS. After that you should have `gcloud` command line tool available and
+authenticated with your google account.
+
+You then need to create a bucket, a service account and get its credentials. You
+can do this via web UI or console. Then you need to put your keys to
+`scripts/ci/gcp-creds.json` and add these to your `ENV`:
+
+```dvc
+$ export GOOGLE_APPLICATION_CREDENTIALS=".gcp-creds.json"
+$ export GCP_CREDS="yes"
+$ export DVC_TEST_GCP_REPO_BUCKET="dvc-test-xyz"
 ```
 
+Here goes a sample code to do this:
+
+```dvc
+# This name needs to be globally unique
+$ export GCP_NAME="dvc-test-xyz"
+$ gcloud projects create $GCP_NAME
+$ gcloud iam service-accounts create $GCP_NAME --project=$GCP_NAME
+$ gcloud iam service-accounts keys create \
+    scripts/ci/gcp-creds.json \
+    --iam-account=$GCP_NAME@$GCP_NAME.iam.gserviceaccount.com
+
+$ gcloud auth activate-service-account \
+    --key-file=scripts/ci/gcp-creds.json
+$ gcloud config set project $GCP_NAME
+$ gsutil mb gs://$GCP_NAME/
+```
+
+I used the same name for project, service account and bucket for simplicity. You
+may use different names.
+
+</details>
+
+<details>
+
+### Click for Azure testing instructions
+
+Install [Node.js](https://nodejs.org/en/download/) and then install and run
+Azurite:
+
+```dvc
+$ npm install -g 'azurite@<3' # Need 2.x version
+$ mkdir azurite
+$ azurite -s -l azurite -d azurite/debug.log
+```
+
+Add this to your `ENV`:
+
+```dvc
+$ export AZURE_STORAGE_CONTAINER_NAME="dvc-test"
+$ export AZURE_STORAGE_CONNECTION_STRING="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;"
+```
```

Comment:
is it a real key?

Reply:
@shcheklein Nah, it is a sample key for a locally installed azurite. No problem in sharing it publically.